packages/integration-tests/tests/social-session.test.ts

```diff
@@ -0,0 +1,121 @@
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { mockSocialConnectorId, mockSocialConnectorConfig } from '@/__mocks__/connectors-mock';
+import {
+  signInWithSocial,
+  getAuthWithSocial,
+  registerWithSocial,
+  bindWithSocial,
+  signInWithUsernameAndPassword,
+} from '@/api';
+import MockClient from '@/client';
+import { setUpConnector, createUserByAdmin } from '@/helpers';
+import { generateUsername, generatePassword } from '@/utils';
+
+const state = 'foo_state';
+const redirectUri = 'http://foo.dev/callback';
+const code = 'auth_code_foo';
+
+describe('social sign-in and register', () => {
+  const userId = crypto.randomUUID();
```

Comment:
The max length of `users.id` in the schema is 12.

Better align with the max length and extract a utility function `generateUserId` in `utils.ts`?

Reply:
it is used as social identity userId

Reply:
Oh. It's the user ID from the social identity provider.

Maybe it should be named as `socialUserId`?

Reply:
updated